src/features/groups/components/GroupSingleHeaderDropdownMenu/index.tsx

```diff
@@ -0,0 +1,76 @@
+'use client';
+import { HeaderMenuCircleButton } from '@/components/parts/Header';
+import { DialogRoot, DrawerRoot, DropdownMenu, DropdownMenuTrigger } from '@/components/ui';
+import { DeleteGroupDialogContent } from '@/features/groups/components/DeleteGroupDialogContent';
+import { useAutoFocus } from '@/hooks';
+import { IGroup } from '@/types/definition';
+
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+
+import { GroupSingleHeaderDropdownMenuContent } from './GroupSingleHeaderDropdownMenuContent';
+import { RenameGroupDrawerContent } from './RenameGroupDrawerContent';
+
+interface IGroupSingleHeaderDropdownMenu {
+  /**
+   * The ID of the group to modify.
+   */
+  groupId: IGroup['id'];
+  /**
+   * The current group name which a user is willing to change
+   */
+  currentGroupName: IGroup['name'];
+}
+
+export const GroupSingleHeaderDropdownMenu = ({
+  groupId,
+  currentGroupName,
+}: IGroupSingleHeaderDropdownMenu) => {
+  const [isRenameGroupDrawerOpen, setIsRenameGroupDrawerOpen] = useState(false);
+  const [isDeleteGroupDialogOpen, setIsDeleteGroupDialogOpen] = useState(false);
```

Comment:
When we need to have more dropdown menu options, consider introduction `useReducer`